lib/open_project/authentication.rb

```diff
@@ -83,7 +83,12 @@ module WWWAuthenticate
       module_function
 
       def pick_auth_scheme(supported_schemes, default_scheme, request_headers = {})
-        req_scheme = request_headers['X-Authentication-Scheme']
+        # Check two keys in case rack or some other middleware translates the
+        # deprecated 'X-Header-Format'. This will be obsolete with rails 4 (and its
+        # more recent rack version) where the deprecated headers will all be mapped to
+        # the HTTP_HEADER_FORMAT.
```

Comment:
If this turns obsolete with Rails 4 I'd suggest to turn this comment into a TODO, that way it _might_ not be forgotten as easily.